src/commands/utils/clusters.ts

```diff
@@ -73,45 +74,180 @@ export function getAksClusterSubscriptionItem(commandTarget: any, cloudExplorer:
+    const aadAccessToken = await getAksAadAccessToken(cluster.subscription.environment, execOptions.serverId, execOptions.tenantId, localToken.refreshToken);
+    if (failed(aadAccessToken)) {
+        return aadAccessToken;
+    }
+
+    // Now we have a token the user can use to access their cluster, but kubelogin doesn't support that being passed as an
+    // argument directly. Instead, we can save it to a temp directory and instruct kubelogin to use that as its cache directory.
+    const cacheDir = storeCachedAadToken(aadAccessToken.result, execOptions);
+    if (failed(cacheDir)) {
+        return cacheDir;
+    }
+
+    // This extension controls the version of kubelogin used, so that:
+    // 1. We don't need to rely on the user having previously downloaded it, and
+    // 2. This workflow doesn't get broken by kubelogin behaviour changes between versions
+    const kubeloginPath = await getKubeloginBinaryPath();
+    if (failed(kubeloginPath)) {
+        return kubeloginPath;
+    }
+
+    // Update the kubeconfig YAML with the new kubelogin options, and return the serialized output as a string.
+    execBlock.command = kubeloginPath.result;
+    execBlock.args = buildExecOptionsWithCache(execOptions, cacheDir.result);
+    const authenticatedKubeConfig = yaml.dump(kubeconfigYaml);
+    return { succeeded: true, result: authenticatedKubeConfig };
+}
+
+interface ExecOptions {
+    subcommand: string,
+    environment: string,
+    serverId: string,
+    clientId: string,
+    tenantId: string,
+    loginMethod: string
+}
+
+function readExecOptions(execArgs: [string]): ExecOptions {
+    // The 'exec' command args are made up of a subcommand (get-token) and a number of name/value options.
+    const [subcommand, ...options] = execArgs;
+
+    // Extract the options into an object for convenient lookup.
+    const optionLookup: { [name: string]: string } = options.reduce((result: { [name: string]: string }, _arg: string, index: number) => {
+        return index % 2 === 0 ?
+            { ...result, [options[index]]: options[index+1] } :
+            result;
+    }, {});
+
+    return {
+        subcommand,
+        environment: optionLookup["--environment"] || optionLookup["-e"],
+        serverId: optionLookup["--server-id"],
+        clientId: optionLookup["--client-id"],
+        tenantId: optionLookup["--tenant-id"],
+        loginMethod: optionLookup["--login"] || optionLookup["-l"]
+    };
+}
+
+function buildExecOptionsWithCache(execOptions: ExecOptions, cacheDir: string) {
+    return [
+        execOptions.subcommand,
+        "--environment", execOptions.environment,
+        "--server-id", execOptions.serverId,
+        "--client-id", execOptions.clientId,
+        "--tenant-id", execOptions.tenantId,
+        // In a non-interactive environment, 'devicecode' login now returns an error, see https://github.com/Azure/kubelogin/pull/145.
```

Comment:
💡☕️ Side not end observation -> This is great information to share, worth for any readers for throwing some light on : `"--login", "interactive",` - Thanks.